sktime/forecasting/base/_base.py

```diff
@@ -266,26 +229,64 @@ def fit_predict(
 
         Parameters
         ----------
-        y : pd.Series
-            Target time series to which to fit the forecaster.
-        fh : int, list, np.array or ForecastingHorizon
-            Forecasting horizon, default = y.index (in-sample forecast)
-        X : pd.DataFrame, optional (default=None)
-            Exogenous time series
+        y : pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+            Time series to which to fit the forecaster.
+            if self.get_tag("scitype:y")=="univariate":
+                must be pd.Series, 1D np.ndarray, or have a single column
+            if self.get_tag("scitype:y")=="multivariate":
+                must be pd.DataFrame or 2D np.ndarray with 2 or more columns
+            if self.get_tag("scitype:y")=="both": no restrictions apply
+        fh : int, list, np.array or ForecastingHorizon (not optional)
+            The forecasters horizon with the steps ahead to to predict.
+        X : pd.DataFrame, or 2D np.array, optional (default=None)
+            Exogeneous data
+            if self.get_tag("X-y-must-have-same-index"),
+            X.index must contain y.index and fh.index
         return_pred_int : bool, optional (default=False)
             If True, returns prediction intervals for given alpha values.
         alpha : float or list, optional (default=0.95)
 
         Returns
         -------
-        y_pred : pd.Series
-            Point predictions
+        y_pred : pd.Series, pd.DataFrame, or np.ndarray (1D or 2D)
+            Point forecasts at fh, with same index as fh
+            y_pred has same type as y
         y_pred_int : pd.DataFrame - only if return_pred_int=True
+            in this case, return is 2-tuple (otherwise a single y_pred)
             Prediction intervals
+
+        State change
```

Comment:
These are not supported headers for the numpy docstring conversation. Could you move them under "Notes"? Same for "Writes to self" (see below)

Reply:
you mean "convention"?
I don't think we settled on a docstring convention, and I think it's important to write the guarantees/assumes here out explicitly, on what the function does.

the `numpy` convention, imo, is mostly for pure functions without side effects and/or being class/object methods.
For class methods, you should document guarantees/assumes for side effects that are specific to object variables.

I don't think "notes" is an appropriate section here.

Can we leave this in - since I think it's important for users and developers - and have a separate discussion about our docstring sections?

Reply:
It generates lots of errors in the docs generation, which makes it really hard to debug it. We've had this discussion during the doc sprint and agreed to rename these sections. So the easiest is to add this under "Notes" as it previously was.

Reply:
Please trust me on software engineering standards for docstrings - in this case I'm pretty sure that the numpy standard is inadequate when it comes to classes and class/object methods.

The "notes" section is for implementation details or background theory.
What I'm talking about is about assumptions/guarantees that are required to document by ["design by contract"](https://en.wikipedia.org/wiki/Design_by_contract) best pracatice.

Reply:
>It generates lots of errors in the docs generation, which makes it really hard to debug it.

Where exactly can I see these errors?

>We've had this discussion during the doc sprint and agreed to rename these sections. 

What I remember is that we agreed to be compliant with the current testing suite (which it doesn't bother?)

As a compromise, how about intead of moving to "notes", we move to "parameters" and "returns", aligning "parameters" with "assumptions" and "returns" with "guarantees"? With explanation if the write/read is with the class.

Reply:
What I want to avoid is that assumptions and side effects are not documented or not documented effectively (e.g., in a section where no one expects or reads them).

Reply:
As said, we've had this discussion before and decided to only use the sections supported by the convention. @RNKuhns spent a lot of time cleaning up the docstrings to make them more compliant. You can see the errors that are generated here: https://readthedocs.org/projects/sktime/builds/14821966/. If we want to add other sections, we should perhaps try to extend the NumPy docstring convention or somehow make sure that we're not generating all these errors.

Reply:
Thanks for pointing me towards the errors - why are these not displayed in the checks? Also, can you tell me what the general process is to see these?

As said above, I'm happy to accommodate by moving the reads/writes to the parameters and results section. What do you think about this alternative suggestion?

Reply:
Don't think that the "Parameters" or "Returns" sections are intended for that info (see https://numpydoc.readthedocs.io/en/latest/format.html).

Reply:
@mloning, can you point me to a resource that explains how I can see the build log from readthedocs?
When I click "details", it directs me to a mockup of the post-PR actual documentation, not to the logs.

Reply:
>Don't think that the "Parameters" or "Returns" sections are intended for that info

Yes, I know - but I think it's more important to have crucial information in the docstring than adherence to a particular documentation style if it doesn't accommodate for full specification of interface contracts.

How about "extended description"? That's a free text section.

Reply:
I mean "extended summary", it sounds like anything can go in there related to functionality, and it would be a good fit for contracts?

Reply:
resolved, formatting changed, numpydoc no longer raises "unknown section" errors